### DISCUSSION TITLE
Unable to extract JSON REST API data, any recommendations

### DISCUSSION THREAD

**Discussion by egliptor** [grafana/grafana-infinity-datasource#60] [Q&A]
Hi,

I am trying to visualise some JSON API data, however when trying to extract the data it’s like the data does not exist and there are no headers to extract from even when inserting the root: Rows.Ad.

Grafana version: 7.5.4
Infinity version: 0.6.0

![Json-02-Grafana](https://user-images.githubusercontent.com/73836983/116010195-76a54a00-a615-11eb-943e-c32bda1628af.png)
![JSON in Browser](https://user-images.githubusercontent.com/73836983/116010196-77d67700-a615-11eb-817a-ffa380c56174.png)

I have tested the example JSON URLs and not pulling through the test data, I'm not sure where I'm going wrong. Any assistance is appreciated.
https://thingspeak.com/channels/38629/feed.json
https://jsonplaceholder.typicode.com/users

Thanks

**Comment by yesoreyeram** [comment]
Providing more details will help. What error you are getting? Is your api behind authentication? How did you setup ur authentication mode ? Etc..

**Reply by egliptor** [reply]
When setting up the Infinity url datasource it states "Failed to fetch data", which I think is due to the JSON output containing the api status, start and end date of the data being called.
When copying the JSON output from the browser into the Infinity JSON inline textbox without the api status and dates the data seems to be fetched. (output pasted below)

The api contains authentication in the url which is compulsory along with the following;
url.com/api/conversions/?key=authentication&camp=0&start=2019-01-01&end=2020-03-01&cols=default

{**"Status":"OK","StartDate":"2020-01-01 00:00:00","EndDate":"2020-03-02 00:00:00",**"Rows":{"Target":[{"Keyword":"","Views":"2","Cost":"0.340","Clicks":"0","CTR":"0.00","Conversion":"0","CR":"0.00","CPA":"0.000","Revenue":"0.00","Profit":"-0.34","ROI":"-100.00"}],"Ad":[{"AdValue":"Ebony 338x235-Image+Text Banner630","Extra1":"","Extra2":"338x235","Extra3":"teen,brunette,heels,stockings","Extra4":"  desktop","Extra5":"403137","Extra6":"{user.browser}","Extra7":"BC","Extra8":"1842687","Extra9":"snippet A Headline 57","Extra10":"snippet B Image 65","Views":"1","EngageRate":"0.00","Cost":"0.170","CPV":"0.170","Clicks":"0","CTR":"0.00","CPC":"0.000","Subscribers":"0","SR":"0.00","CPSUB":"0.000","Conversion":"0","CR":"0.00","CPA":"0.000","Revenue":"0.00","EPV":"0.000","PPV":"-0.170","Profit":"-0.17","eCPM":"-170.00","ROI":"-100.00"},{"AdValue":"Ebony 338x235-Image+Text Banner630","Extra1":"","Extra2":"338x235","Extra3":"teen,brunette,heels,stockings","Extra4":"  desktop","Extra5":"403137","Extra6":"{user.browser}","Extra7":"BC","Extra8":"1842687","Extra9":"","Extra10":"","Views":"1","EngageRate":"0.00","Cost":"0.170","CPV":"0.170","Clicks":"0","CTR":"0.00","CPC":"0.000","Subscribers":"0","SR":"0.00","CPSUB":"0.000","Conversion":"0","CR":"0.00","CPA":"0.000","Revenue":"0.00","EPV":"0.000","PPV":"-0.170","Profit":"-0.17","eCPM":"-170.00","ROI":"-100.00"}],"Landing":[{"LandingPageID":"9","LP":"Ebony Milfs-L1-Lander11-N4-BC","Extra1":"","Extra2":"338x235","Extra3":"teen,brunette,heels,stockings","Extra4":"  desktop","Extra5":"403137","Extra6":"{user.browser}","Extra7":"BC","Extra8":"1842687","Extra9":"","Extra10":"","Views":"1","EngageRate":"0.00","Cost":"0.170","CPV":"0.170","Clicks":"0","CTR":"0.00","CPC":"0.000","Conversion":"0","CR":"0.00","CPA":"0.000","Revenue":"0.00","EPV":"0.000","PPV":"-0.170","Profit":"-0.17","eCPM":"-170.00","ROI":"-100.00"},{"LandingPageID":"9","LP":"Ebony Milfs-L1-Lander11-N4-BC","Extra1":"","Extra2":"338x235","Extra3":"teen,brunette,heels,stockings","Extra4":"  desktop","Extra5":"403137","Extra6":"{user.browser}","Extra7":"BC","Extra8":"1842687","Extra9":"snippet A Headline 57","Extra10":"snippet B Image 65","Views":"1","EngageRate":"0.00","Cost":"0.170","CPV":"0.170","Clicks":"0","CTR":"0.00","CPC":"0.000","Conversion":"0","CR":"0.00","CPA":"0.000","Revenue":"0.00","EPV":"0.000","PPV":"-0.170","Profit":"-0.17","eCPM":"-170.00","ROI":"-100.00"}],"Landing2":[],"Landing3":[],"Landing4":[],"Landing5":[]}}

**Reply by egliptor** [reply]
To obtain the above JSON output from the url API would the below root structure be heading in the right direction? At present I keep getting "Failed to fetch data"

json.StartDate[0].EndDate[0].Rows.Ad

**Reply by yesoreyeram** [reply]
@egliptor did you manage to get it working or having issues? in browser, check network tab to see if you are getting response.

**Reply by egliptor** [reply]
I'm still having issues getting it to pass into Infinity with a URL that has a JSON output.

Testing the URL in the browser displays the data like in the image originally posted.
Not sure what you mean check with the developer 'Network Tab' as the URL is displaying the data and status ok.

**Reply by yesoreyeram** [reply]
Sorry @egliptor that you are having trouble getting data.

I mean though the URL returns data, due to connectivity from grafana server or any misconfiguration of the datasource may cause the error.

What is the "URL" setting in the datasource config and in the panel?  Also I would recommend to get the latest version of the plugin.
